kafka.go

```diff
@@ -636,3 +685,50 @@ func (client *KafkaClient) runCallback(
 		Key:              event.Key,
 	}, nil)
 }
+
+func (client *KafkaClient) doPublish(ctx context.Context, topic string, config kafka.WriterConfig,
```

Comment:
this method seems similar to publishEvent, any reason we can't reuse it?

Reply:
yes, I think we can reuse it